File: day01/main.js
import fs from "fs";

function main() {
    const input = fs.readFileSync("inputs/day01.txt", "utf8");
    const lines = input.split("\n");
    const left = [];
    const right = [];
    let sum = 0;

    for (let line of lines) {
        const [a, b] = line.split("   ").map(Number);
        left.push(a);
        right.push(b);
    }

    left.sort((a, b) => a - b);
    right.sort((a, b) => a - b);

    for (let i = 0; i < left.length; i++) {
        sum += left[i] > right[i] ? left[i] - right[i] : right[i] - left[i];
    }

    console.log(sum); // 1222801
}

main();

File: day01/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"path/filepath"
	"sort"
	"strconv"
	"strings"
)

func check(e error) {
	if e != nil {
		log.Fatal(e)
	}
}

func main() {
	cwd, err := os.Getwd();
	check(err)

	path := filepath.Join(cwd, "inputs", "01.txt")

	file, err := os.Open(path)
	check(err)
	defer file.Close()

	scanner := bufio.NewScanner(file)
	
	var left []int
	var right []int
	
	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.Fields(line)
		a, _ := strconv.Atoi(parts[0])
		b, _ := strconv.Atoi(parts[1])
		left = append(left, a)
		right = append(right, b)
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}

	sort.Ints(left)
	sort.Ints(right)

	sum := 0
	for i := range left {
		if left[i] > right[i] {
			sum += left[i] - right[i]
		} else {
			sum += right[i] - left[i]
		}
	}

	fmt.Println(sum) // 1222801
}
